speech_recognition/__init__.py

```diff
@@ -1702,6 +1702,47 @@ def recognize_vosk(self, audio_data, language='en'):
         
         return finalRecognition
 
+    def recognize_speechmatics(self, audio_data, key=None, language="en", transcript_format="txt"):
+        """
+        Performs speech recognition on ``audio_data`` (an ``AudioData`` instance), using the Speechmatics ASR
+
+        The key value is your speechmatics API key. You can get an API key by creating an account and signing into the portal at https://portal.speechmatics.com/manage-access/.
+
+        The recognition language is determined by ``language``, an RFC5646 language tag like "en" or "es". The full list of supported languages can be found at https://docs.speechmatics.com/introduction/supported-languages.
+
+        Returns a text representation of the transcript by default. You can alson get a json representation of the transcript by setting transcript_format='json-v2', which comes with a range of meta-data about each word in the transcript. The full transcript schema is documented here: https://docs.speechmatics.com/features. You can also request an SRT format by setting `format='srt'`
+
+        Raises a ``speech_recognition.UnknownValueError`` exception if the speech is unintelligible. Raises a ``speech_recognition.RequestError`` exception if the speech recognition operation failed, if the key isn't valid, or if there is no internet connection.
+        """
+        assert isinstance(audio_data, AudioData), "Data must be audio data"
+        assert isinstance(key, str), "``key`` must be a string"
+
+        try:
+            from speechmatics.models import ConnectionSettings, BatchTranscriptionConfig
+            from speechmatics.batch_client import BatchClient
+            from speechmatics.constants import BATCH_SELF_SERVICE_URL
+        except:
+            raise RequestError("missing speechmatics python module: install using `pip install speechmatics-python`")
+
+        wav_data = audio_data.get_wav_data(
+            convert_rate=None if audio_data.sample_rate >= 16000 else 16000  # audio samples must be at least 16 kHz
```

Comment:
You might know better than me - is this correct? Do we deal with sample rate >= 16000 ourselves?

Reply:
I asked Dan, he reckons we don't get anything less than 8000, but may be able to support lower than that. It's not documented anywhere that I can see. Maybe we just let the runtime deal with it and pass any errors to users?

Reply:
Yeah sounds good to me, we can keep an eye on the repo and see if anyone is having issues if it gets merged

Reply:
Removed that now, will resolve and merge and then open a PR into the upstream